### ISSUE TITLE
Add API to expose generated properties 

### ISSUE THREAD

**Issue by brempusz** [n0mer/gradle-git-properties#102]
Could you expose generated properties as map or provide some api for accessing generated values? I would like to use generated values in other tasks. As a workaround I do this in `build.gradle`:

```
ext {
    gitProps = new Properties()
}
// ...
task loadGitProperties(type: Task) {
    dependsOn tasks.generateGitProperties
    doLast {
        generateGitProperties.output.withInputStream { InputStream istr -> project.gitProps.load(istr) }
    }
}
// ...
// in some task other task:
"${-> gitProps['git.branch'] ?: 'not-set'}"
```

Just `gitProperties['git.branch']` or `gitProperties.find('git.branch')` without additional custom code would make more sense.

**Comment by tha2015** [comment]
Thanks for the suggestion. I'm thinking about exposing a Map as task output so the properties can be accessed as `generateGitProperties.properties['git.commit.id.abbrev']`
BTW because of the UP-TO-DATE check mechanism of Gradle, sometimes, the task will not be executed and so the Map would not be exposed, you will have to force task execution with
```
generateGitProperties.outputs.upToDateWhen { false }
```
And of course you will still need 
```
dependsOn tasks.generateGitProperties
```
to make sure `generateGitProperties` to be executed before your logic

I'm also thinking about adding `generateGitPropertiesFile` config for the users who needs the properties (but not the generated git.properties file). But it would complicate the way up-to-date check works for the plugin.

What do you think?

**Comment by brempusz** [comment]
I don't see any benefits in having just exposed properties without the file. Gained performance boost will be negligible but it will introduce more complicated code.

**Comment by tha2015** [comment]
This feature should be supported in the next release v1.5.0. Specifically, `extProperty` config key will be able to be configure to expose the properties map under project.ext

**Comment by sdavids** [comment]
Please also add a task to display all generated properties.

**Comment by tha2015** [comment]
@sdavids  If it is for debugging purpose, I would suggest to use plugin version 1.5 with "--debug" option to display the generated map (see https://github.com/n0mer/gradle-git-properties/blob/master/src/main/groovy/com/gorylenko/GenerateGitPropertiesTask.groovy). If it is not for the debugging, why do you think the extProperty config will not sufficient for the job?

**Comment by tha2015** [comment]
Closing because the feature is now available in v1.5.0
Please open new ticket for other requests. Thanks.
